marklogic-data-hub/src/main/java/com/marklogic/hub/mapping/MappingImpl.java

```diff
@@ -49,6 +53,25 @@ public MappingImpl(String name) {
         this.targetEntityType = "http://example.org/modelName-version/entityType";
     }
 
+    public MappingImpl(String name, HubEntity entity) {
```

Comment:
I think a unit test would be useful here and fairly easy to write as well, as there's what looks like some important logic. 

The test could also verify that the new "getEntityFromProject" method works as expected.